Source/SwiftLintBuiltInRules/Rules/Lint/PrivateSwiftUIStatePropertyRule.swift

```diff
@@ -202,58 +32,170 @@ private extension PrivateSwiftUIStatePropertyRule {
         }
 
         /// LIFO stack that stores type inheritance clauses for each visited node
-        /// The last value is the inheritance clause for the most recently visited node
-        /// A nil value indicates that the node does not provide any inheritance clause
-        private var visitedTypeInheritances = Stack<InheritanceClauseSyntax?>()
+        /// `true` indicates the the most recently visited decl syntax 
+        /// should be evaluated for SwiftUI State access levels
+        private var visitedScopes = Stack<Bool>()
 
         override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
-            visitedTypeInheritances.push(node.inheritanceClause)
+            visitedScopes.push(node.inheritanceClause.conformsToApplicableSwiftUIProtocol)
             return .visitChildren
         }
 
         override func visitPost(_ node: ClassDeclSyntax) {
-            visitedTypeInheritances.pop()
+            visitedScopes.pop()
         }
 
         override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
-            visitedTypeInheritances.push(node.inheritanceClause)
+            visitedScopes.push(node.inheritanceClause.conformsToApplicableSwiftUIProtocol)
             return .visitChildren
         }
 
         override func visitPost(_ node: StructDeclSyntax) {
-            visitedTypeInheritances.pop()
+            visitedScopes.pop()
         }
 
         override func visit(_ node: ActorDeclSyntax) -> SyntaxVisitorContinueKind {
-            visitedTypeInheritances.push(node.inheritanceClause)
+            visitedScopes.push(node.inheritanceClause.conformsToApplicableSwiftUIProtocol)
             return .visitChildren
         }
 
         override func visitPost(_ node: ActorDeclSyntax) {
-            visitedTypeInheritances.pop()
+            visitedScopes.pop()
+        }
+
+        override func visit(_ node: VariableDeclSyntax) -> SyntaxVisitorContinueKind {
+            guard let parent = node.parent, parent.is(MemberBlockItemSyntax.self) else {
```

Comment:
This condition needs to go into `visitPost`. Please have a test for it, e.g.:
```swift
struct MyView: View {
    var s: Int {
        @State // This would trigger
        var s: Int = 3
        return s
    }

    var body: some View { Text("") }
}
```

Not sure this is valid. But currently it would trigger and `private` is not allowed there.